packages/shared/src/hooks/translation/useTranslation.ts

```diff
@@ -0,0 +1,141 @@
+import { useCallback, useEffect, useRef } from 'react';
+import type { InfiniteData, QueryKey } from '@tanstack/react-query';
+import { useQueryClient } from '@tanstack/react-query';
+import { stream } from 'fetch-event-stream';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { apiUrl } from '../../lib/config';
+import type { FeedData, Post } from '../../graphql/posts';
+import {
+  updateCachedPagePost,
+  findIndexOfPostInData,
+  updatePostCache,
+} from '../../lib/query';
+
+export enum ServerEvents {
+  Connect = 'connect',
+  Message = 'message',
+  Disconnect = 'disconnect',
+  Error = 'error',
```

Comment:
Error handling missing? Not sure if we want to do anything

Reply:
Not sure if we need to handle anything on client side, only thing I can think of is to retry.

Reply:
In the meantime, can add log and metric how often errors occur on kvasir side to see what types of errors might occur. So far I haven't really managed to get any errors from Translate API